In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import urllib.request

Problemas 2.0.1:
Escriba una funci´on que calcule el factorial de n, con n ∈ N
$$ n!=n(n-1)(n-2)(n-3)...$$
Calcule los primeros 20 n´umeros factoriales.

In [ ]:
def factorial (n):
  y=0
  x=1
  if n==0 or n==1:
      y=1
  else:
    while n>1:
        x*=n
        n-=1
        y=x     
  return y
for i in range(20):
  print(factorial(i))
        

1
1
2
6
24
120
720
5040
40320
362880
3628800
39916800
479001600
6227020800
87178291200
1307674368000
20922789888000
355687428096000
6402373705728000
121645100408832000


Escriba una funcion que calcule las variaciones sin repeticion de n elemenos tomados de
r en r:
$$ V_{r}^{n}=\frac{n!}{(n-r)!}$$
a) Calcule de cuantas maneras puedo ubicar 6 carros en 3 estacionamientos. Ans: 120


In [ ]:
def variacion (n,r):
  return factorial(n)/factorial(n-r)
print(variacion(6,3))

120.0


Escriba una funcion que calcule las combinaciones sin repeticion de n elementos tomados
de m en m, con n > m.
$$ C_{m}^{n}=\frac{n!}{m!(n-m)!}$$
Calcule cuantos equipos de 11 jugadores puedo formar con 22 jugadores disponibles.
Suponga que:


a) Cualquiera puede ser el arquero. Ans: 705432


b) Ya sabemos quien sera el arquero. Ans: 352716

In [ ]:
def combinacion (n,m):
  if m>0 and m<n:
    C=variacion(n,m)/factorial(m)
  return C
print(combinacion(22,11))
print(combinacion(21,10))

705432.0
352716.0
